api/types/trait/convert/v1/trait.go

```diff
@@ -25,15 +25,21 @@ import (
 
 // FromProto converts an array of v1 traits into a map of string to string array.
 func FromProto(traits []*traitv1.Trait) trait.Traits {
+	if traits == nil {
```

Comment:
Do you rely on traits being nil vs being empty?

In Protobuf, there is no distinction between empty and nil arrays, as both are represented the same way in the proto wire format. Specifically, Protobuf serializes both empty and nil arrays as the absence of the field in the wire format, meaning that when the data is deserialized, it cannot determine whether the original array was empty or nil. 

This design choice simplifies their serialization process but requires careful handling to ensure that the intended meaning (empty vs. nil) is preserved.

Reply:
> Do you rely on traits being nil vs being empty?

There are many places where `cmp.Equals` is called  that distinguish between `array == nil and len(array) == 0

Right now for the same struct after conversation the result is incorrect.